src/T8code.jl

```diff
@@ -212,6 +212,13 @@ mutable struct ForestWrapper
     end
 end
 
+Base.pointer(fw::ForestWrapper) = fw.pointer
+# This dispatched conversion allows to conveniently pass a ForestWrapper object
+# to `t8_forest_...` calls.  The following link leads to a lengthy discussion
+# about if this is a valid way to achieve this:
+# https://discourse.julialang.org/t/how-to-use-ccall-cconvert-and-unsafe-convert-in-a-convenient-and-memory-safe-way/41932/18
+Base.unsafe_convert(::Type{Ptr{T8code.Libt8.t8_forest}}, fw::ForestWrapper) = fw.pointer
```

Comment:
That's really helpful. For the `PointerWrapper`s we use something like
```julia
function Base.unsafe_convert(::Type{Ptr{T8code.Libt8.t8_forest}}, fw::ForestWrapper)
    Base.unsafe_convert(Ptr{T8code.Libt8.t8_forest}, pointer(fw))
end
```
cf. https://github.com/trixi-framework/P4est.jl/blob/7f151d751da34ab747fb2c8cab2e7830c0f4dca4/src/pointerwrappers.jl#L54-L56
This doesn't make any difference in this case because the additional `unsafe_convert` does not do anything (`pointer(fw)` is already a `Ptr{T8code.Libt8.t8_forest}`), right?

Reply:
Yes, indeed.